datafusion/core/src/catalog/mod.rs

```diff
@@ -53,28 +53,32 @@ pub trait CatalogList: Sync + Send {
     fn catalog(&self, name: &str) -> Option<Arc<dyn CatalogProvider>>;
 }
 
+/// See [`CatalogProviderList`]
```

Comment:
👍  -- this will probably still error on compilation, but at least it should point users in the right way 👍